ReText/editor.py

```diff
@@ -242,15 +242,25 @@ def handleReturn(self, cursor):
 		cursor.movePosition(QTextCursor.StartOfBlock, QTextCursor.KeepAnchor)
 		text = cursor.selectedText()
 		length = len(text)
-		pos = 0
-		while pos < length and (text[pos] in (' ', '\t')
-		  or text[pos:pos+2] in ('* ', '- ', '> ')):
-			pos += 1
-		if pos == length:
-			cursor.removeSelectedText()
-		# Reset the cursor
-		cursor = self.textCursor()
-		cursor.insertText(('\n' + text[:pos]) if pos < length else '\n')
+		match = re.search("^[\\s]*([*>-]|\\d+\\.) ", text)
```

Comment:
Please pre-compile the regexes for performance using re.compile.

Reply:
Done :+1: